### PR TITLE
chore(webpage): add id to homepage in h2

### PR DESCRIPTION
**Pull request by kmin-jeong**
## PR Checklist

- [x] Addresses an existing open issue: #5949 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

### CODE REVIEW

File: packages/website/src/pages/index.tsx [hunk 1]
```diff
@@ -108,7 +108,7 @@ const features: FeatureItem[] = [
 function Feature({ title, description }: FeatureItem): JSX.Element {
   return (
     <div className="col col--12 padding-vert--lg">
-      <h2 className="text--center">{title}</h2>
+      <h2 id="title" className="text--center">{title}</h2>
```
Comment (Josh-Cena) [inline packages/website/src/pages/index.tsx:111]: Docusaurus has a `Heading` component. In order for the clickable "#" to show up you have to use that. This isn't documented, but check out https://github.com/facebook/docusaurus/blob/b4c04f28f760cb7ea9de709643df4c23f8b0e934/website/src/pages/versions.tsx#L64

**Review by Josh-Cena — CHANGES_REQUESTED** [review]

**Comment by Josh-Cena** [comment]
You are missing the import. You can check the file I linked.

**Comment by kmin-jeong** [comment]
@Josh-Cena  I keep getting errors on the lintcheck, is there any separate command to install lint?

File: packages/website/src/pages/index.tsx [hunk 2]
```diff
@@ -108,7 +109,7 @@ const features: FeatureItem[] = [
 function Feature({ title, description }: FeatureItem): JSX.Element {
   return (
     <div className="col col--12 padding-vert--lg">
-      <h2 className="text--center">{title}</h2>
+      <Heading as="h2" id="title" className="text--center">{title}</Heading>
```
Comment (JoshuaKGoldberg) [inline packages/website/src/pages/index.tsx:112]: I like the idea of using the theme's `Heading` component, +1!

But, it's not centered visually on desktop:

Screenshot of a left-aligned heading above centered text

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for sending this in @kmin-jeong! I noted a visual bug in a comment.

A couple of requests:

* We normally ask that folks read the contributing guide and fill out the PR template in full: https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md#pull-requests
* If the PR isn't ready for review yet, please [put it in draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request)

File: packages/website/src/pages/index.tsx (hunk 2 above)
Comment (kmin-jeong) [reply packages/website/src/pages/index.tsx:112]: @JoshuaKGoldberg 
If there is an installation command in the website folder, where can I find it? I tried the yarn install, but Docusaurus doesn't seem to work. Is there anything I missed?

Comment (JoshuaKGoldberg) [reply packages/website/src/pages/index.tsx:112]: > If there any extra installation command in the website folder

Hmm, no, the way Yarn workspaces work, running `yarn` anywhere should be fine.

> tried the yarn install, but Docusaurus doesn't seem to work

Tip: it helps to be very specific on what you've done and what's happening. By _"doesn't seem to work"_ you're not giving much information. Someone trying to help (me!) can't be specific in diagnosing your issue. Help us help you!

It also helps to say what you've read through. Have you seen the docs at https://github.com/typescript-eslint/typescript-eslint/blob/75dcdf164d206c5530ba7cc095c4599ec90abe35/DEVELOPMENT.md? What happens when you go through them?

Comment (kmin-jeong) [reply packages/website/src/pages/index.tsx:112]: @JoshuaKGoldberg  I complete that the installation is finished with the yarn install command. When I execute the yarn build command, I get the same error as shown in the picture.
build error

Comment (Josh-Cena) [reply packages/website/src/pages/index.tsx:112]: Did you run `yarn install` at the root of the repo?

**Comment by kmin-jeong** [comment]
@JoshuaKGoldberg  I re-request a review of my pr.I separated the heading tag and className. I put <P> tag with {discription}. I checked test have passed like the picture.
pass

File: packages/website/src/pages/index.tsx [hunk 3]
```diff
@@ -107,9 +108,13 @@ const features: FeatureItem[] = [
 
 function Feature({ title, description }: FeatureItem): JSX.Element {
   return (
-    <div className="col col--12 padding-vert--lg">
-      <h2 className="text--center">{title}</h2>
-      {description}
+    <div className="col col--12 padding-vert--lg ">
```
Comment (JoshuaKGoldberg) [inline packages/website/src/pages/index.tsx:111]: Nit: lil extra space added

```suggestion
    <div className="col col--12 padding-vert--lg">
```

File: packages/website/src/pages/index.tsx [hunk 4]
```diff
@@ -107,9 +108,13 @@ const features: FeatureItem[] = [
...
+      <div className="text--center">
+        <Heading as="h2" id="title">
+          {title}
+        </Heading>
+      </div>
```
Comment (JoshuaKGoldberg) [inline packages/website/src/pages/index.tsx:116]: Heh, I filed https://github.com/facebook/docusaurus/issues/8327 that we should be able to put the `className="text--center"` on the `Heading` (but can't right now).

File: packages/website/src/pages/index.tsx [hunk 5]
```diff
@@ -107,9 +108,13 @@ const features: FeatureItem[] = [
 
 function Feature({ title, description }: FeatureItem): JSX.Element {
   return (
-    <div className="col col--12 padding-vert--lg">
-      <h2 className="text--center">{title}</h2>
-      {description}
+    <div className="col col--12 padding-vert--lg ">
+      <div className="text--center">
+        <Heading as="h2" id="title">
```
Comment (JoshuaKGoldberg) [inline packages/website/src/pages/index.tsx:113]: > `id="title"`

There are multiple headings rendered by this component. [Element IDs should be unique within the entire document](https://stackoverflow.com/questions/5611963/can-multiple-different-html-elements-have-the-same-id-if-theyre-different-eleme). Linking to the `#title` hash shouldn't be ambiguous as to which heading it links to.

You'll want the `id` prop here to be computed based on the `title`. Keep in mind that you'll need to replace any spaces with some character like `-`. I'd suggest also lowercasing it. 

A suggestion for your own testing of the PR: make sure you can click the `#` next to the headings. And, when you click that, make sure it doesn't navigate to a _different_ heading in the same page.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Getting closer! Visually it looks fine, though the hash linking is a bit off.

A suggestion in general: you might want to spend five minutes before re-requesting review trying to play around with the changes like a real user. See what website behaviors are now different because of your code changes, and make sure that those behaviors all work as a user would expect.

(this is a funny suggestion coming from me because up through around 2020, I was _notorious_ in my professional teams for not doing that)

File: packages/website/src/pages/index.tsx (hunk 3 above)
Comment (kmin-jeong) [reply packages/website/src/pages/index.tsx]: I fixed it!

File: packages/website/src/pages/index.tsx (hunk 4 above)
Comment (kmin-jeong) [reply packages/website/src/pages/index.tsx:116]: So I moved the class name to div, should I modify it?

File: packages/website/src/pages/index.tsx (hunk 5 above)
Comment (kmin-jeong) [reply packages/website/src/pages/index.tsx]: @JoshuaKGoldberg 
I modified the `title` to `start-type script-eslint-title`. I checked that the tag next to `Why does this project exists?` is moved to `What are ESLint and TypeScript, and how do they compare?`, do you want it not to move?

File: packages/website/src/pages/index.tsx (hunk 4 above)
Comment (JoshuaKGoldberg) [reply packages/website/src/pages/index.tsx:116]: Oh sorry I didn't mean to imply you need to do anything. This is perfect. Docusaurus just needs to fix a bug (or tell me I'm wrong about it being a bug 😂).

**Comment by Josh-Cena** [comment]
(Reminder to remove `feat` from the PR title)

File: packages/website/src/pages/index.tsx [hunk 6]
```diff
@@ -108,8 +109,12 @@ const features: FeatureItem[] = [
 function Feature({ title, description }: FeatureItem): JSX.Element {
   return (
     <div className="col col--12 padding-vert--lg">
-      <h2 className="text--center">{title}</h2>
-      {description}
+      <div className="text--center">
+        <Heading as="h2" id="start-typescript-eslint-title">
```
Comment (JoshuaKGoldberg) [inline packages/website/src/pages/index.tsx:113]: @kmin-jeong this isn't right either. Let me try to explain from the start.

This `Feature` function component is rendered multiple times on the page. Each place it's rendered will need a unique ID.

The point of the ID is that someone can use it to link to that particular heading in the page. For example, `https://typescript-eslint.io/docs/#details` links to the heading in https://typescript-eslint.io/docs that has `id="details"`.

Right now it's hardcoded to have the same ID each time: `"start-typescript-eslint-title"`. Which is not good: if two headings have the same ID, linking to the URL for that ID could mean either of them. This is what I meant by "ambiguous" in https://github.com/typescript-eslint/typescript-eslint/pull/5951#discussion_r1020767451.

All that being said, I tried this out locally and as it turns out, the `Heading` component automatically adds a good `id` by default! So we don't need to provide one at all.

```suggestion
        <Heading as="h2">
```

That should simplify things. 😄

File: packages/website/src/pages/index.tsx [hunk 7]
```diff
@@ -161,7 +166,9 @@ function Home(): JSX.Element {
         ))}
         <section className={styles.sponsors}>
           <div className="container text--center padding-vert--lg">
-            <h2>Financial Contributors</h2>
+            <Heading as="h2" id="financial-contributors">
```
Comment (JoshuaKGoldberg) [inline packages/website/src/pages/index.tsx:169]: (see comment above)

```suggestion
            <Heading as="h2">
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Almost there!

File: packages/website/src/pages/index.tsx (hunk 5 above)
Comment (JoshuaKGoldberg) [reply packages/website/src/pages/index.tsx]: Sorry, I don't follow your explanation here. I tried out the latest version and it's equivalently not working.

But https://github.com/typescript-eslint/typescript-eslint/pull/5951#discussion_r1021995099 should fix things! ✨

File: packages/website/src/pages/index.tsx (hunk 6 above)
Comment (Josh-Cena) [reply packages/website/src/pages/index.tsx:113]: > automatically adds a good `id` by default!

I'd be surprised—I don't think we've added that logic 😕 If there's no `id`, it should render a heading with no `#` link at all.

Comment (JoshuaKGoldberg) [reply packages/website/src/pages/index.tsx:113]: Huh, you're right! Sorry about that! I must have been looking at a cached version or something. I'll edit my post.

Comment (Josh-Cena) [reply packages/website/src/pages/index.tsx:113]: Yeah, sometimes the hot reload lags behind. Always a good idea to refresh.

Comment (kmin-jeong) [reply packages/website/src/pages/index.tsx]: @JoshuaKGoldberg  ok! I modified `"start-typescript-eslint-title"` to `id = {title}`.
I used regex in replace() and toLowerCase(). It replace the space or `( , )` or `( _ )` with a `"-"` .

Comment (kmin-jeong) [reply packages/website/src/pages/index.tsx]: + I updated the branch and pulled it to the feat/5949 branch, but the following error occurs.I'm looking for an issue, but I don't see a clear solution. I use window10 and visual studio code.
errors

Comment (JoshuaKGoldberg) [reply packages/website/src/pages/index.tsx]: Oh yeah sorry that was me, unrelated. Fixed in #6001.

Comment (kmin-jeong) [reply packages/website/src/pages/index.tsx]: after updated branch,It's resolved!

**Review by JoshuaKGoldberg — APPROVED** [review]
Looks great to me! Thanks for sending this in & working with us on it @kmin-jeong! 🎉  

@Josh-Cena you've got more Docusaurus knowledge than me and I have an instinct I might be missing something again, so I'll defer to you if you have time soon to take a look. 😁

**Review by Josh-Cena — APPROVED** [review]
LGTM

**Pull request merged**